modules/v1/workspaces/models/ArticleFile.php

```diff
@@ -0,0 +1,164 @@
+            [['content'], 'string'],
+            [['name', 'label', 'mime'], 'string', 'max' => 255],
+            [['path'], 'string', 'max' => 512],
+            [['article_id'], 'exist', 'skipOnError' => true, 'targetClass' => Article::class, 'targetAttribute' => ['article_id' => 'id']],
+            [['created_by'], 'exist', 'skipOnError' => true, 'targetClass' => User::class, 'targetAttribute' => ['created_by' => 'id']],
+            [['updated_by'], 'exist', 'skipOnError' => true, 'targetClass' => User::class, 'targetAttribute' => ['updated_by' => 'id']],
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function attributeLabels()
+    {
+        return [
+            'id' => Yii::t('app', 'ID'),
+            'article_id' => Yii::t('app', 'Article ID'),
+            'name' => Yii::t('app', 'Name'),
+            'label' => Yii::t('app', 'Label'),
+            'path' => Yii::t('app', 'Path'),
+            'mime' => Yii::t('app', 'Mime'),
+            'content' => Yii::t('app', 'Content'),
+            'size' => Yii::t('app', 'Size'),
+            'to_process' => Yii::t('app', 'To Process'),
+            'processing' => Yii::t('app', 'Processing'),
+            'created_at' => Yii::t('app', 'Created At'),
+            'updated_at' => Yii::t('app', 'Updated At'),
+            'created_by' => Yii::t('app', 'Created By'),
+            'updated_by' => Yii::t('app', 'Updated By'),
+        ];
+    }
+
+    /**
+     * @return ArticleFileQuery
+     */
+    public static function find()
+    {
+        return new ArticleFileQuery(get_called_class());
+    }
+
+    /**
+     * Gets query for [[Article]].
+     *
+     * @return ActiveQuery
+     */
+    public function getArticle()
+    {
+        return $this->hasOne(Article::class, ['id' => 'article_id']);
+    }
+
+    /**
+     * Gets query for [[CreatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getCreatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'created_by']);
+    }
+
+    /**
+     * Gets query for [[UpdatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getUpdatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'updated_by']);
+    }
+
+    /**
+     * Upload file
+     *
+     * @param UploadedFile $file
+     * @return bool
+     * @throws Exception
+     */
+    public function uploadFile(UploadedFile $file)
+    {
+        $path = '/article-attachments/' . $this->article_id;
+        $fullPath = Yii::getAlias('@storage' . $path);
+
+        $this->name = $file->name;
+        $this->path = $path;
+        $this->mime = $file->type;
+        $this->size = $file->size;
+
+        if (!file_exists($fullPath)) if (!FileHelper::createDirectory($fullPath)) {
+            throw new ValidationException(Yii::t('app', "Directory '$fullPath' was NOT created"));
```

Comment:
@skavt It is not good idea to show full path in error, because you return it to frontend and user might probably read it. You can save it in error logs but not in such exception which can be accessible by users.